### ISSUE TITLE
Exposing alertmanager through Kubernetes api and alert urls

### ISSUE THREAD

**Issue by tsloughter** [coreos/prometheus-operator#342]
**What did you do?**

Used `kube-prometheus` deploy script and then followed https://coreos.com/operators/prometheus/docs/latest/user-guides/exposing-prometheus-and-alertmanager.html#kubernetes-api for exposing prometheus and alertmanager.

**What did you expect to see?**

Links in alertmanager generated alerts that point to the kubernetes proxy url. Updating the `externalUrl` of the `AlertManager` resource probably does update the links that are contained in alerts to point to the right url but I am unable to verify since configuring the `externalUrl` causes no alerts to be fired.

**What did you see instead? Under which circumstances?**

No alerts when `externalUrl` of `AlertManager` is set.

**Environment**

```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:33:11Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

GKE

* Manifests:

```
apiVersion: "monitoring.coreos.com/v1alpha1"
kind: "Alertmanager"
metadata:
  name: "main"
  labels:
    alertmanager: "main"
spec:
  replicas: 3
  version: v0.6.1
```

and:

```
apiVersion: "monitoring.coreos.com/v1alpha1"
kind: "Alertmanager"
metadata:
  name: "main"
  labels:
    alertmanager: "main"
spec:
  replicas: 3
  version: v0.6.1
  externalUrl: https://127.0.0.1:8001/api/v1/proxy/namespaces/monitoring/services/alertmanager-main:web/
```

**Comment by brancz** [comment]
Interesting... this may require us to either also implement route-prefix for alertmanager or be able to provide an endpoint prefix in the alertmanager endpoint struct @fabxc

**Comment by brancz** [comment]
@tsloughter which links are you referring to? Should these links be pointing at the Prometheus UI or the Alertmanger UI? Because if it's the Prometheus UI then you need to properly configure the `externalUrl` in the Prometheus resource.

**Comment by tsloughter** [comment]
@brancz links to the alertmanager ui. I am using the prometheus `externalUrl` successfully, it has the `routePrefix` option that I had to also set for it to work.

**Comment by brancz** [comment]
Yes that was my suspicion. We'll investigate how to solve this best, but I can't guarantee this to happen very quickly as we'll have to take this upstream, as route prefix does not exist for the Alertmanager itself yet.

**Comment by brancz** [comment]
`-web.route-prefix` has been added to upstream Alertmanager and released with `v0.7.0`. The version used in this repo has been updated and the exposition docs have been adapted.

I tested all types of access: `kubectl proxy`, `kubectl port-forward` and via ingress and all for consistently. Let me know if this works for you as well.

I will close this here.
